applications/ShapeOptimizationApplication/tests/sensitivity_analysis_test/run_semi_analytic_step_size_verification.py

```diff
@@ -0,0 +1,83 @@
+    "sensitivity_settings" : {
+        "sensitivity_model_part_name"     : "Parts_structure",
+        "nodal_sensitivity_variables"     : ["SHAPE_SENSITIVITY"],
+        "element_sensitivity_variables"   : [],
+        "condition_sensitivity_variables" : [],
+        "build_mode": "static"
+    }
+}""")
+
+list_of_move_nodes = [5, 65]
+finite_difference_levels = [2,3,4,5,6,7,8,9,10,11,12]
+results_filename = "fd_semianalytic_results.txt"
+
+# ==============================================================================
+# Preprocessing
+# ==============================================================================
+with open(kratos_response_settings["primal_settings"].GetString(),'r') as primal_parameters:
+    primal_parameters = KM.Parameters(primal_parameters.read())
+model_part_name = primal_parameters["solver_settings"]["model_part_name"].GetString()
+domain_size = primal_parameters["solver_settings"]["domain_size"].GetInt()
+
+open_file = open(results_filename,"w")
+open_file.close()
+
+# ==============================================================================
+# Analysis
+# ==============================================================================
+for move_node_id in list_of_move_nodes:
+
+    # Initialize results file
+    with open(results_filename,'a') as open_file:
+        open_file.write("-----------------------------------------------------------------------\n")
+        open_file.write("time\t\t\t= "+str(time.ctime())+"\n")
+        open_file.write("move_node_id\t= "+str(move_node_id)+"\n")
+        open_file.write("---\n")
+        open_file.write("step,\tgradient_x,\tgradient_y,\tgradient_z,\ttime\n")
+
+    for itr in finite_difference_levels:
+
+        print("\n\n##########################################################################")
+        print(">> Starting finite difference level ",itr, "for node ", str(move_node_id))
+        print("##########################################################################\n\n")
+
+        # Perturb
+        current_delta = 1*10**(-itr)
+        kratos_response_settings["step_size"].SetDouble(current_delta)
+
+        # Run simulation
+        model = KM.Model()
+        response = structural_response_function_factory.CreateResponseFunction(kratos_response_settings["response_type"].GetString(), kratos_response_settings, model)
+        response.RunCalculation(True)
+
+        # Write results
+        gradient_of_interest = response.GetShapeGradient()[move_node_id]
+        with open(results_filename,'a') as open_file:
+            line_to_write = '%.0E' % Decimal(str(current_delta)) + ",\t"
+            line_to_write += '%.6E' % Decimal(str(gradient_of_interest[0])) + ",\t"
+            line_to_write += '%.6E' % Decimal(str(gradient_of_interest[1])) + ",\t"
+            line_to_write += '%.6E' % Decimal(str(gradient_of_interest[2])) + ",\t"
+            line_to_write += str(time.ctime()) + "\n"
+            open_file.write(line_to_write)
```

Comment:
This is useful but it is not a test - meaning it will not be tested and it will become outdated quite easily.

Reply:
Maybe we can make another folder with some useful skripts inside?

Reply:
I believe those scripts can be really useful in the verification of new response functions. But you are right, the problem is that they are not tested. Actually here it is not about a single verification, but only about whether or not the script runs through and so follows the latest Kratos standards. 

So, I am thinking to add the three scripts to the nightly tests such that it is frequently tested if the process is still running. What do you think? The execution time is only a few seconds.

Reply:
That is fine for me.